src/utils/document-section-utils.ts

```diff
@@ -0,0 +1,268 @@
+        const sectionElements = proseMirrorContainer.children;
+        const collectionArray = Array.from(sectionElements);
+        const nodeIndexs = getIndexBySectionName(collectionArray, nodeStyle);
+        const startingIndex = nodeIndexs[workingSection];
+        const sectionLevel = getStyleLevel(nodeStyle, storedStyles);
+
+        if (!sectionLevel || !sectionElements[startingIndex]) break;
+
+        sectionElements[startingIndex].remove();
+
+        if (!sectionElements[startingIndex]) break;
+
+        deleteDocumentChildElements(startingIndex, sectionElements, storedStyles);
+      }
+
+      const removedSection = tempTocNodeList.findIndex(node => node.attrs.objectId === id);
+      tempTocNodeList.splice(removedSection, 1);
+    }
+  }
+
+  return renderedDoc;
+}
+
+export function buildListOfIdsToAdd(
+  sectionId: string,
+  currentListOfExcludedIds: string[],
+  flatStructure: FlatSectionNodeStructure[]
+): string [] {
+  const section = flatStructure.find(section => section.id === sectionId);
+  section.isChecked = true;
+
+  const ids = addSelectedSection(flatStructure, sectionId);
+  let newNodeList = JSON.parse(JSON.stringify(currentListOfExcludedIds));
+
+  for (const id of ids) {
+    newNodeList = newNodeList.filter(nodeId => nodeId !== id);
+  }
+
+  return sortExcludeListByFlattenedSection(newNodeList, flatStructure);
+}
+
+export function buildListOfIdsToRemove(
+  sectionId: string,
+  currentListOfExcludedIds: string[],
+  flatStructure: FlatSectionNodeStructure[]
+): string[] {
+  let newNodeList = JSON.parse(JSON.stringify(currentListOfExcludedIds));
+  const section = flatStructure.find(section => section.id === sectionId);
+  section.isChecked = false;
+
+  const allNewIds = getAllSectionIds(section, flatStructure);
+
+  newNodeList = [...newNodeList, ...allNewIds];
+  newNodeList = [...new Set(newNodeList)];
+
+  return sortExcludeListByFlattenedSection(newNodeList, flatStructure);
+}
+
+function sortExcludeListByFlattenedSection(nodeList: string[], flatStructure: FlatSectionNodeStructure[]): string[] {
+  if (nodeList.length) {
+    nodeList.sort((a, b) => {
+      return flatStructure.findIndex(section => section.id === a) - flatStructure.findIndex(section => section.id === b);
+    });
+  }
+
+  return nodeList;
+}
+
+function getAllSectionIds(section: FlatSectionNodeStructure, flatStructure: FlatSectionNodeStructure[]): string[] {
+  let allChildIds = [];
+
+  if (section.childrenIds.length) {
+    for (const id of section.childrenIds) {
+      const childSection = flatStructure.find(section => section.id === id);
+      const nestedIds = getAllSectionIds(childSection, flatStructure);
+      allChildIds = [...allChildIds, ...nestedIds];
+    }
+  }
+
+  return [section.id , ...allChildIds];
+}
+
+function toggleDisableInput(id: string, isChecked: boolean): void {
+  const uniqueSectionId = `licit-pdf-export-${id}`;
+  const input = document.getElementById(uniqueSectionId) as HTMLInputElement;

```

Comment:
document.get... is an inherently unsafe operation, especially within React which will redraw elements, and bundles UI updates together. You should update the data that feeds the HTML, and leave all HTML edits to React.

Reply:
We have to toggle programmatically and state was not working as intended when I tried to use it do disable the inputs. This was the next simplest solution.